### PR TITLE
Default weight to 1 for unweighted graph during n2v

### LINKED ISSUE #800: [BUG] type error when running node2vec_embed
## Expected Behavior
Generate a node2vec embedding of the input graph

## Actual Behavior
`TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'` occurs during the _preprocess_transition_probabilities() step in node2vec_embed. I'm not sure why, but unnormalized_probs is currently a list of all 'None's.

## Example Code
```python
g = nx.erdos_renyi_graph(10, 0.7, directed=False)
tensors, labels = node2vec_embed(g, dimensions=128, num_walks=10, window_size=2, iterations=3)
```
The parameters are from layouts\auto.py

## Full Traceback
```pytb
 File "C:\Users\d0156\OneDrive\Documents\GitHub\graspologic\graspologic\layouts\test_n2v.py", line 5, in <module>
    tensors, labels = node2vec_embed(g)
  File "C:\Users\d0156\OneDrive\Documents\GitHub\graspologic\graspologic\embed\n2v.py", line 116, in node2vec_embed
    node2vec_graph._preprocess_transition_probabilities()
  File "C:\Users\d0156\OneDrive\Documents\GitHub\graspologic\graspologic\embed\n2v.py", line 437, in _preprocess_transition_probabilities
    norm_const = sum(unnormalized_probs)
TypeError: unsupported operand type(s) for +: 'int' and 'NoneType'
```

## Your Environment
* Python version: 3.9

## Additional Details
Same error occurs for directed graphs too.

### PR DESCRIPTION
**Pull request by Nyecarr**
- [ ] Does this PR add any new dependencies?
- [x] Does this PR modify any existing APIs?
   - [x] Is the change to the API backwards compatible?
- [ ] Have you built the documentation (reference and/or tutorial) and verified the generated documentation is appropriate?

#### What does this implement/fix? Briefly explain your changes.
Embedding methods should handle unweighted graphs by defaulting the weight to 1. N2v was not behaving in the same way as the spectral methods -- this PR will fix that.

Closes #800 

#### Any other comments?
I also added an explicity list() when returning the labels with a corresponding test as networkx graph.nodes() returns a NodeView rather than a list.

### CODE REVIEW

File: graspologic/embed/n2v.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def node2vec_embed(
         f"Completed. Ending time is {str(end)} Elapsed time is {str(start - end)}"
     )
 
-    labels = node2vec_graph.original_graph.nodes()
+    labels = list(node2vec_graph.original_graph.nodes())
```
Comment (Nyecarr) [inline graspologic/embed/n2v.py:133]: NodeView issue, return a list instead.

File: graspologic/utils/utils.py [hunk 2]
```diff
@@ -974,7 +977,7 @@ def remap_node_ids(
     node_id_dict = dict()
     graph_remapped = type(graph)()
 
-    for source, target, weight in graph.edges(data=weight_attribute):
+    for source, target, weight in graph.edges(data=weight_attribute, default=weight_default):
```
Comment (Nyecarr) [inline graspologic/utils/utils.py:980]: Use specified weight if a weight attribute is not found for a given edge.

File: graspologic/embed/n2v.py [hunk 3]
```diff
@@ -13,17 +13,17 @@
...
-    walk_length: int = 80,
-    return_hyperparameter: float = 1.0,
-    inout_hyperparameter: float = 1.0,
-    dimensions: int = 128,
-    window_size: int = 10,
-    workers: int = 8,
-    iterations: int = 1,
-    interpolate_walk_lengths_by_node_degree: bool = True,
-    random_seed: Optional[int] = None,
+        graph: Union[nx.Graph, nx.DiGraph],
```
Comment (dwaynepryce) [inline graspologic/embed/n2v.py:16]: why do you always tab so far in

File: graspologic/utils/utils.py [hunk 4]
```diff
@@ -945,7 +945,9 @@ def remap_labels(
 
 
 def remap_node_ids(
-    graph: nx.Graph, weight_attribute: str = "weight"
+        graph: nx.Graph,
+        weight_attribute: str = "weight",
+        weight_default: int = 1
```
Comment (dwaynepryce) [inline graspologic/utils/utils.py:950]: make this a float, not an int

File: graspologic/utils/utils.py (hunk 2 above)
Comment (dwaynepryce) [reply graspologic/utils/utils.py:980]: do an nx.is_weighted() on it first and warn on no weight - if they expected their graph to be weighted and mistyped the weight attribute on add or something, they won't even know we're ignoring a weight

File: graspologic/embed/n2v.py (hunk 3 above)
Comment (Nyecarr) [reply graspologic/embed/n2v.py:16]: Default pep8-ifyer in intellij.

Comment (dwaynepryce) [reply graspologic/embed/n2v.py:16]: I can't believe they'd do this, just waste all of our precious characters when we _only get 88_.  it's an outrage, Jetbrains.

File: graspologic/utils/utils.py [hunk 5]
```diff
@@ -971,10 +975,16 @@ def remap_node_ids(
     if not isinstance(graph, nx.Graph):
         raise TypeError("graph must be of type nx.Graph")
 
+    if not nx.is_weighted(graph, weight=weight_attribute):
+        logger = logging.getLogger("graspologic.utils")
```
Comment (dwaynepryce) [inline graspologic/utils/utils.py:979]: I was thinking the warnings.warn module actually.

Comment (Nyecarr) [reply graspologic/utils/utils.py]: Updated to use warnings module

File: graspologic/utils/utils.py [hunk 6]
```diff
@@ -1030,10 +1031,17 @@ def remap_node_ids(
     if not isinstance(graph, nx.Graph):
         raise TypeError("graph must be of type nx.Graph")
 
+    if not nx.is_weighted(graph, weight=weight_attribute):
+        warnings.warn(
+            f'Graph is unweighted using weight_attribute "{weight_attribute}". Defaulting weights to "{weight_default}"'
```
Comment (daxpryce) [inline graspologic/utils/utils.py:1036]: Maybe change it to:

`f'Graph has at least one unweighted edge using weight_attribute "{weight_attribute}". Defaulting unweighted edges to "{weight_default}"'`.  Then we're very precise.

**Review by daxpryce — APPROVED** [review]

**Pull request merged**